internal/logger/audit.go

```diff
@@ -183,13 +184,22 @@ func AuditLog(ctx context.Context, w http.ResponseWriter, r *http.Request, reqCl
 			statusCode      int
 			timeToResponse  time.Duration
 			timeToFirstByte time.Duration
+			outputBytes     int64 = -1 // -1: unknown output bytes
 		)
 
-		st, ok := w.(*ResponseWriter)
-		if ok {
+		var st *ResponseWriter
+		switch v := w.(type) {
+		case *ResponseWriter:
+			st = v
+		case *gzhttp.GzipResponseWriter:
+			// the writer may be obscured by gzip response writer
+			st = v.ResponseWriter.(*ResponseWriter)
```

Comment:
Lets remove the unchecked assert:
```suggestion
			if rw, ok = v.ResponseWriter.(*ResponseWriter); ok {
				st = rw
			}
```